serial.go

```diff
@@ -49,10 +51,10 @@ type serialhub struct {
 	mu sync.Mutex
 }
 
-// SpPortList is the serial port list
-type SpPortList struct {
-	Ports []SpPortItem
-	Mu    sync.Mutex `json:"-"`
+// SerialPortList is the serial port list
+type SerialPortList struct {
+	Ports     []*SpPortItem
```

Comment:
```suggestion
    ports     []*SpPortItem
```
This was public only because it was needed by the json Marshal function under the `List` method.

Reply:
Actually no, it really wants an object like:

```json
{ "ports": [ ..... ] }
```

(even if in the console you see `[ ]` in reality the raw response is an object with the "ports" field).